woocommerce/woocommerce
# Marketplace: hide subscriptions with maxed connections and fix "not connected" badge

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

1) WooCommerce.com subscriptions have a limit for connections. When that is reached, users can no longer connect that subscription to a store. This PR hide those subscriptions from the My Subscriptions page.
2) We show a "not connected" badge on the "available to use" table which is not ideal and confusing. It should be only shown in the "installed" table. 
3) Similarly with the "update" button. It should only be shown in the "installed" table. 

Closes https://github.com/Automattic/woocommerce.com/issues/19499.



### How to test the changes in this Pull Request:

1. Checkout this branch and get a new build
2. Connect your local WooCommerce dev environment to WooCommerce.test and connect your store.
    - I use this patch to do it: [connect-to-local.zip](https://github.com/user-attachments/files/17093957/connect-to-local.zip)
    - You may also need to bridge the Docker containers using `docker network connect woocommerce.test <IN_APP_WP_CONTAINER_ID>`
4. Find a paid product that you have two subscriptions to. Ideally with different expiry dates, so you can tell which is which (although this is not mandatory). 
5. Install the paid product. I have used the manual method and uploaded the zip. Visit [the My Subscriptions page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions) and one of the subscription items should now appear in the **Installed on this store** table and the other should remain in the **Available to use** table.
6. Confirm the item in **Available to Use** doesn't have a status badge in the **Subscription** column about missing connection. Previously this was a bug and we showed this badge.
7. Apply this patch (please extract from the zip first) to change the latest version of your product so that the locally installed version is out of date. Update the product slug and version accordingly.
8. Confirm there is an update button on the **Installed on this store** table but not one on the **Available to use** table
Patch: [product-version.zip](https://github.com/user-attachments/files/17093912/product-version.zip)



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Fix bugs in the Marketplace > My Subscription page

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

![image](https://github.com/user-attachments/assets/479ef4e5-aa82-44cd-9d95-10906db3b4ad)
